src/util/Injected.js

```diff
@@ -362,19 +388,43 @@ exports.LoadUtils = () => {
             isNewMsg: true,
             type: 'chat',
             ...ephemeralFields,
+            ...mediaOptions,
+            ...(mediaOptions.toJSON ? mediaOptions.toJSON() : {}),
+            ...quotedMsgOptions,
             ...locationOptions,
             ..._pollOptions,
-            ...attOptions,
-            ...(attOptions.toJSON ? attOptions.toJSON() : {}),
-            ...quotedMsgOptions,
             ...vcardOptions,
             ...buttonOptions,
             ...listOptions,
             ...extraOptions
         };
 
+        if (isChannel) {
+            const msg = new window.Store.Msg.modelClass(message);
+            const msgDataFromMsgModel = window.Store.SendChannelMessage.msgDataFromMsgModel(msg);
+            const isMedia = Object.keys(mediaOptions).length > 0;
+            await window.Store.SendChannelMessage.addNewsletterMsgsRecords([msgDataFromMsgModel]);
+            chat.msgs.add(msg);
+            chat.t = msg.t;
+
+            const sendChannelMsgResponse = await window.Store.SendChannelMessage.sendNewsletterMessageJob({
+                msg: msg,
+                type: message.type === 'chat' ? 'text' : isMedia ? 'media' : 'pollCreation',
+                newsletterJid: chat.id.toJid(),
```

Comment:
@shirser121 
> Maybe we should allow some extra option so pepole can control some fileds

Unfortunately, once you you change those props, the message is not being sent.
But I have not tested it a lot.
You can try to play with adding some extra properties. If it will work for you, just suggest your changes and I'll add them as your commit :)

Reply:
@shirser121 
Don't we already have extra options [here](https://github.com/alechkos/whatsapp-web.js/blob/05e9eab099bdf1a312a78aa45edb72ee2ebdc46c/src/util/Injected.js#L399)?